packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js

```diff
@@ -0,0 +1,201 @@
+    return {
+      hideOperators: true,
+      values: this.typeOptions.reduce(
+        (result, { label, value }) => ({
+          ...result,
+          [value]: label,
+        }),
+        {},
+      ),
+    };
+  }
+
+  get canCreateAccess() {
+    return (
+      this.accessForm?.$valid &&
+      !this.isCreatingAccess &&
+      !this.isLoadingAccessForm
+    );
+  }
+
+  get canShowAccessForm() {
+    return (
+      !this.isAccessFormShown &&
+      !this.isLoadingAccesses &&
+      !this.isLoadingAccessForm
+    );
+  }
+
+  loadAccesses($config) {
+    const { criteria, pageSize, offset, sort } = $config;
+
+    let accessesQuery = this.iceberg(`${this.partitionApiUrl}/access`)
+      .query()
+      .expand('CachedObjectList-Pages')
+      .limit(pageSize)
+      .offset(offset)
+      .sort(sort.property, sort.dir > 0 ? 'ASC' : 'DESC');
+
+    criteria.forEach(({ property, value, operator }) => {
+      accessesQuery = accessesQuery.addFilter(
+        property,
+        ...{
+          bigger: ['gt', value],
+          contains: ['like', `%${value}%`],
+          containsNot: ['nlike', `%${value}%`],
+          endsWith: ['like', `%${value}`],
+          is: ['eq', value],
+          isAfter: ['gt', value],
+          isBefore: ['lt', value],
+          isNot: ['nq', value],
+          smaller: ['lt', value],
+          startsWith: ['like', `${value}%`],
+        }[operator],
+      );
+    });
+
+    this.isLoadingAccesses = true;
+    this.hideAccessForm();
+
+    return accessesQuery
+      .execute(null, true)
+      .$promise.catch(({ data: error }) => {
+        this.alertError(error);
+        return { data: [] };
+      })
+      .then(({ data }) => {
+        this.accesses = data;
+        if (!this.accesses.length) {
+          return this.showAccessForm().then(() => this.accesses);
+        }
+        return this.accesses;
+      })
+      .then((data) => ({
+        data,
+        meta: { totalCount: data.length },
+      }))
+      .finally(() => {
+        this.isLoadingAccesses = false;
+      });
+  }
+
+  showAccessForm() {
+    let promise = this.$q.when();
+
+    if (!this.authorizedAccesses) {
+      const { partitionApiUrl } = this;
+
+      this.isLoadingAccessForm = true;
+
+      promise = this.$q
+        .all({
+          ips: this.$http.get(`${partitionApiUrl}/authorizableIps`),
+          blocks: this.$http.get(`${partitionApiUrl}/authorizableBlocks`),
+        })
+        .then(({ ips: { data: ips }, blocks: { data: blocks } }) => {
+          this.authorizedAccesses = [
+            ...ips
+              .filter((x) => !this.accesses.find(({ ip }) => ip === `${x}/32`))
+              .map((ip) => ({ type: 'IP', ip })),
+            ...blocks
+              .filter((block) => !this.accesses.find(({ ip }) => ip === block))
+              .map((block) => ({ type: 'IP/Block', ip: block })),
+          ].sort(
+            ({ ip: ipBlockA }, { ip: ipBlockB }) =>
+              Number(
```

Comment:
Shouldn't this be extracted to a function ? transformation is the same and maybe it could simplify `showAccessForm` logic  slightly

Reply:
:+1: